x-pack/plugins/security_solution/server/endpoint/routes/metadata/index.ts

```diff
@@ -106,4 +107,23 @@ export function registerEndpointRoutes(
     },
     getMetadataRequestHandler(endpointAppContext, logger)
   );
+
+  router.get(
+    {
+      path: `/api/endpoint/toAgentID`,
+      validate: { query: schema.object({ ids: schema.string() }) },
+    },
+    async (context, req, res) => {
+      const agents = await GetAgentIDsForEndpoints(
+        req.query.ids.split(','),
```

Comment:
why not make `ids` an array?

Reply:
what does the URL look like when you do that? same param multiple times?

Reply:
the query string in this case is like `?ids=a,b,c`

Turning it into an array becomes.. interesting. Even sending a single value `?ids=a` gets a JSON validation error. I think it immediately expects a `["a","b"]` format _in the query string_. Which is nonsense.